architecture/architecture.puml

```diff
@@ -0,0 +1,39 @@
+﻿@startuml
+
+package "AV Datalogger" {
+    [Connector] as Connector
+    [Data Store] as Store
+    [Car Logger] as Log
+    [App Logger] as AppLog
+    [App] as App
+
+    (CSV File) as CSV
+    (Serial Port) as Serial
+    (TCP Socket) as TCP
+    
+    CSV <-- Connector : reads from
+    Serial <-down- Connector : reads from
+    TCP <-down- Connector : reads from
+
+    Connector -right-> Store : converts and stores
+
+    Store -right-> Log : logs CAN data
+    App -up-> Store : reads from
+    App --> Connector : sends commands
+    App -right-> AppLog : logs app data
+
+    Log --> [CSV File] : writes logs
+    AppLog  --> [TXT File] : writes logs
```

Comment:
What is the purpose of the App Logs being a TXT file instead of a CSV file?

Reply:
I'd argue it allows us to have a less structured log format, if it were to be CSV it would only have headers like `time,level,log`.
I would also like to think that we won't be programatically parsing the app log idk

Reply:
That is fair. If we do end up doing the stretch goal of report generation, would we be using the App Logs?

Reply:
I think we would have some report generator component, it would be done independetly to the app logs. In my mind app logs are primarily for debugging / monitoring purposes, not for functionality, but my mind isn't set any way here idm

Reply:
Okay, that is cool with me, happy for it to remain as TXT file